### PR TITLE
feat(compactor2): add catalog update information to tests

### PR DESCRIPTION
**Pull request by alamb**
re https://github.com/influxdata/idpe/issues/17161

# Rationale
As suggested by @NGA-TRAN  in https://github.com/influxdata/influxdb_iox/pull/7026#discussion_r1118831725

In order to make the layout tests clearer, we would like to know when the compactor has decided to upgrade existing files (rather than rerun the plans and make new files). This information is already implicit in output of the test (by the id assignments of the files) but it would be better if the action was explicit in the test output

# Changes:
1. Add a way to intercept catalog commit calls,
2. Install a component that records catalog updates (only in test mode)
3. Update simulator to append to same catalog trace
4. Update test output to show the catalog upgrades

### CODE REVIEW

File: compactor2/tests/layouts/knobs.rs [hunk 1]
```diff
@@ -208,6 +217,9 @@ async fn all_overlapping_l0_split_percentage_and_max_file_size() {
     - "L2.?[172,184] 10.8mb                                                         |-L2.?--|              "
     - "L2.?[184,196] 10.8mb                                                                   |-L2.?--|    "
     - "L2.?[196,200] 3.6mb                                                                             |L2.?|"
+    - "Committing partition 1:"
```
Comment (alamb) [inline compactor2/tests/layouts/knobs.rs:220]: Specifically, this output now answers the question @NGA-TRAN  had at https://github.com/influxdata/influxdb_iox/pull/7026/files#r1118831725 about if the files were upgraded or rewritten

(they are rewritten)

File: compactor2_test_utils/src/commit_wrapper.rs [hunk 2]
```diff
@@ -0,0 +1,103 @@
+//! Handles recording commit information to the test run log
```
Comment (alamb) [inline compactor2_test_utils/src/commit_wrapper.rs:1]: This file contains the ceremony needed to install a wrapper around the `Commit` reference created in the `Components`

File: compactor2_test_utils/src/simulator.rs [hunk 3]
```diff
@@ -54,47 +60,17 @@ impl std::fmt::Display for ParquetFileSimulator {
...
         Self {
-            runs: Arc::new(Mutex::new(vec![])),
+            run_log,
+            run_id_generator: AtomicUsize::new(0),
         }
     }
 
-    /// Get a visual display of the simulated runs this simulator
-    /// has performed, for use in tests
-    pub fn runs(&self) -> Vec<String> {
```
Comment (alamb) [inline compactor2_test_utils/src/simulator.rs:66]: the formatting code is moved down

File: compactor2/tests/layouts/knobs.rs (hunk 1 above)
Comment (alamb) [reply compactor2/tests/layouts/knobs.rs:220]: BTW @NGA-TRAN  I didn't see any "Upgrading" messages -- thus I conclude that none of our tests ever actually upgrade files (they only rewrite them)

**Review by NGA-TRAN — APPROVED** [review]
This makes sense. Thanks @alamb. After handling large size, I will add related tests and investigate why we do not see upgrade. If you can help open a ticket, it would be great, Or I will open the ticket when I start working on it.

File: compactor2/tests/layouts/knobs.rs (hunk 1 above)
Comment (alamb) [reply compactor2/tests/layouts/knobs.rs:220]: Filed https://github.com/influxdata/influxdb_iox/issues/7087 to track

**Comment by alamb** [comment]
> This makes sense. Thanks @alamb. After handling large size, I will add related tests and investigate why we do not see upgrade. If you can help open a ticket, it would be great, Or I will open the ticket when I start working on it.

Thanks! I filed https://github.com/influxdata/influxdb_iox/issues/7087 to track

**Pull request merged**